App.js

```diff
@@ -11,42 +11,55 @@ const chevronStyle = {width: 30, height: 30, resizeMode: 'contain', marginRight:
 const chatTitleStyle = {fontSize: 18, fontWeight: 'bold'};
 const titleStyle = {fontSize: 24, fontWeight: 'bold', flex: 1};
 
+const keysBlocklist = ['key', 'stale', 'routeNames']
+
+const stripNavigationState = (state) => {
+  const rawState = Array.isArray(state) ? [] : {};
+  for (const key of Object.keys(state)) {
+    if (keysBlocklist.includes(key)) {
+      continue
+    }
+    rawState[key] = typeof state[key] === 'object' ? stripNavigationState(state[key]) : state[key]
+  }
+  return rawState 
+}
+
```

Comment:
This function is used to remove `keys` and `stale` from the navigation state so we can regenerate it. Keeping the same `keys` for both navigators causes some problems. BTW not sure about `routeNames` it may be ok to leave them.

Reply:
`routeNames` seems like it wouldn't change. But it also seems like something we wouldn't need to include in an `initialState` or state reset.

Reply:
I'm not too sure.

Reply:
I'm also a little unsure what the "stale" things in the state are used for (even after reading [this](https://reactnavigation.org/docs/navigation-state/#partial-state-objects)) but it seems like something we might not need to worry about.

Reply:
To be extra safe it makes sense to get rid of the `stale` key or set it to `true` otherwise react-navigation will not fix any issue in the state object.

**From the doc:**
```
If you want React Navigation to fix invalid state, you need to make sure that you don't have stale: false in the state object. State objects with stale: false are assumed to be valid state objects and React Navigation won't attempt to fix them.
```
---------------------------------------------------------------------------
Also based on the above assumption that react-navigation will handle fixing state for us we can get rid of `routeNames`.

**From the doc:**
you can only provide a routes array without any keys and React Navigation will automatically add everything that's needed to make it work:
```js
const state = {
  routes: [{ name: 'Home' }, { name: 'Profile' }],
};
```
After rehydration, it'll look something like this:
```js
const state = {
  type: 'stack',
  key: 'stack-1',
  routeNames: ['Home', 'Profile', 'Settings'],
  routes: [
    { key: 'home-1', name: 'Home' },
    { key: 'settings-1', name: 'Settings' },
  ],
  index: 1,
  stale: false,
};
```
Here, React Navigation filled in the missing bits such as keys, route names, index etc.

Reply:
Thanks @techievivek, good notes on the rehydrated state. I think that answers whether we need `routeNames` or `key`.

Next, I agree that we could remove the `stale` property. But my point above is that I am not sure whether we **need to**. 

If the `state` is "stale" then it sounds like we might not want to use it at all? I'd like to understand more about whether the `state` will ever be missing `stale: false` when passing through this function before we make an effort to remove it. Otherwise, we are just doing stuff without a clear picture of why it matters.

Reply:
Like this...

>If you want React Navigation to fix invalid state, you need to make sure that you don't have stale: false in the state object

When do we want React Navigation to fix the "invalid state" and what does it mean? Do we want it to do this or not?